src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,11 +1,25 @@
 package core.basesyntax.service;
 
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (storageDao.get(user.getLogin()) == null && user.getLogin() != null
+                && user.getPassword() != null
+                && user.getPassword().matches("^\\w*\\d+\\w*$")
+                && user.getPassword().matches("^\\w*[A-Z]+\\w*$")
```

Comment:
let's use one regex

Reply:
done. if you know a shorter way to do it please tell me